core/src/main/scala/tech/beshu/ror/acl/Acl.scala

```diff
@@ -36,7 +38,15 @@ object AclHandlingResult {
   object Result {
     final case class Allow(blockContext: BlockContext, block: Block) extends Result
     final case class ForbiddenBy(blockContext: BlockContext, block: Block) extends Result
-    case object ForbiddenByUnmatched extends Result
+    final case class ForbiddenByMismatched(causes: NonEmptySet[Cause]) extends Result
+    case object ForbiddenByMismatched {
+      sealed trait Cause
+      object Cause {
+        case object ActionNotAllowed extends Cause
```

Comment:
Here the "Action" word is generic for the whole set of characteristics of the request? i.e. not the action as in the actions rule. Right?

Reply:
yes. Maybe it should be called OperationNotAllowed

Reply:
Love it. Yes, let's change it and merge.